File: Week-1/Task2.c
// Suppose you are implementing a Linear search algorithm using any of the languages like C/Python. In the given array if there is a possibility of multiple occurrences of some elements. In such a case how to identify the location of the element. Design and implement the solution for the same.
// Sample input
// Enter number of elements 6
// Enter 6 elements
// 44,16,18,16,47,16
// Enter element to search 16
// Sample output
// 16 is present at location 1
// 16 is present at location 3
// 16 is present at location 5

// Program(C):
#include<stdio.h>
int main(){
int a[100],i,found=0,n,key;
printf("Enter number of elements:");
scanf("%d",&n);
printf("Enter %d elements",n);
for(i=0;i<=n-1;i++)
scanf("%d",&a[i]);
printf("Enter element to search:");
scanf("%d",&key);
for(i=0;i<=n-1;i++)
if(a[i]==key){
    printf("%d is present at location %d\n",key,i);
    found=1;
}
if(found==0)
printf("Element is not found");
return 0;
}

// Output 1:
// Enter number of elements:5
// Enter 5 elements 1 2 3 4 2 2
// Enter element to search:2
// 2 is present at location 1
// 2 is present at location 4
// Output 2:
// Enter number of elements:6
// Enter 6 elements 1 54 3 32 5 -5
// Enter element to search:4
// Element is not found

File: Week-4/Task2.py
# We have discussed sorting a algorithm in the class. Implement the same algorithm using any of the languages like C/Python. Follow the following LOC.
# Sample Input and Output
# Enter number of Elements
# 6
# Enter array of elements
# 12,45,10,33,55,50
# Sorted array is
# 10,12,33,45,50,55

# Program(Python):
def partition(a,l,r):
    pivot=a[r]
    i=l-1
    for j in range(l,r+1):
        if(a[j]<pivot):
            i+=1
            a[i],a[j]=a[j],a[i]
    a[i+1],a[j]=a[j],a[i+1]
    return i+1

def quicksort(a,l,r):
    if(l<r):
        pi=partition(a,l,r)
        quicksort(a,l,pi-1)
        quicksort(a,pi+1,r)
    return a

n=int(input("Enter number of Elements "))
a= list(map(int,input("Enter the array of elements:").split()))[:n]
print(quicksort(a,0,len(a)-1))

# Output 1:
# Enter number of Elements 8
# Enter the array of elements:1 4 2 6 3 8 11 7
# [1, 2, 3, 4, 6, 7, 8, 11]
# Output 2:
# Enter number of Elements 5
# Enter the array of elements:1 3 2 1 1
# [1, 1, 1, 2, 3]
